vars/solutionsCFNTest.groovy

```diff
@@ -0,0 +1,94 @@
+def call(Map config = [:]) {
+
+    pipeline {
+        agent { label config.workerAgent ?: 'Jenkins-Default-Agent-X64-C5xlarge-Single-Host' }
+
+        parameters {
+            string(name: 'GIT_REPO_URL', defaultValue: 'https://github.com/lewijacn/opensearch-migrations.git', description: 'Git repository url')
+            string(name: 'GIT_BRANCH', defaultValue: 'sol-pipeline', description: 'Git branch to use for repository')
+            string(name: 'STAGE', defaultValue: "sol-integ", description: 'Stage name for deployment environment')
+        }
+
+        options {
+            // Acquire lock on a given deployment stage
+            lock(label: params.STAGE, quantity: 1, variable: 'stage')
+            timeout(time: 1, unit: 'HOURS')
+            buildDiscarder(logRotator(daysToKeepStr: '30'))
+        }
+
+        stages {
+            stage('Checkout') {
+                steps {
+                    script {
+                        git branch: "${params.GIT_BRANCH}", url: "${params.GIT_REPO_URL}"
+                    }
+                }
+            }
+
+            stage('Deployment') {
+                steps {
+                    timeout(time: 15, unit: 'MINUTES') {
+                        dir('deployment/migration-assistant-solution') {
+                            script {
+                                env.STACK_NAME_SUFFIX = "${stage}-us-east-1"
+                                sh "sudo npm install"
+                                withCredentials([string(credentialsId: 'migrations-test-account-id', variable: 'MIGRATIONS_TEST_ACCOUNT_ID')]) {
+                                    withAWS(role: 'JenkinsDeploymentRole', roleAccount: "${MIGRATIONS_TEST_ACCOUNT_ID}", region: "us-east-1", duration: 3600, roleSessionName: 'jenkins-session') {
+                                        sh "sudo --preserve-env cdk deploy 'Migration-Assistant-Infra-Create-VPC' --parameters Stage=${stage} --require-approval never --concurrency 3"
```

Comment:
use `npm deploy` to reduce extra parameters like `--require-approval` and CDK doesn't need to be installed

Reply:
This only saves me not having to provide the `--require-approval` flag I believe and CDK CLI will already be on the box. My preference is to keep the same, but open if others want to change it.

Reply:
I'd really prefer we didn't have to run `npm install -g cdk` as part of the env setup, by using npm run it ensures everything is correctly version locked. 🧂